### DISCUSSION TITLE
Can agnn.py runnable with 'mag' or 'dblp' dataset

### DISCUSSION THREAD

**Discussion by pdhung3012** [pyg-team/pytorch_geometric#3764] [Q&A]
The current agnn.py, in my understanding, is the implementation of a homogeneous GNN model for planetoid dataset. When I check this dataset, it only has nodes with no information about type and also the edges between nodes.

My question is, can the current agnn.py be available for support other datasets that have different types of nodes such as 'mag' and 'dblp'? To do that, do we need to remove information about types of nodes in these datasets?

Thank you

**Comment by rusty1s** [comment]
You should be able to utilize [`to_hetero`](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.to_hetero_transformer.to_hetero) to convert an `AGNN` network into its heterogeneous counterpart, see [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/to_hetero_mag.py#L42-L49). Let me know if this works for you :)
